### ISSUE TITLE
[FEATURE] databricks_aws_s3_mount encryption

### ISSUE THREAD

**Issue by davehowell** [databrickslabs/terraform-provider-databricks#801]
### Terraform Version
```
Terraform v1.0.4
on darwin_arm64
+ provider registry.terraform.io/databrickslabs/databricks v0.3.7
+ provider registry.terraform.io/hashicorp/aws v3.53.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- databricks_aws_s3_mount
- databricks_mws_workspaces

### Expected Behavior
Customer data buckets should be encrypted either globally or per bucket.

### Actual Behavior
There is no way to set encryption on customer data buckets globally in the workpace. It is also not possible per bucket as the `databricks_aws_s3_mount` resource has no attribute or block for this.

### Important Factoids

I set up customer provided keys, and then realized that feature only covers the root & log buckets. The docs recommend not storing actual data in the root bucket, so I have configured data buckets. But then there's no way to set up server-side encryption using the terraform provider.

The doc -> https://docs.databricks.com/data/data-sources/aws/amazon-s3.html#write-files-using-sse-s3
says: 
> Set up global KMS encryption properties in a AWS configurations setting OR using an init script.

That would be ideal, if it could be done globally. I already have instance profiles, and a KMS key. Those manual docs mention adding the instance profile as a key user ( I think it should be the IAM Role that is allowed to use the key, not the instance profile but that's not important). 

The docs also show these examples for manually mounting buckets with encryption
```
dbutils.fs.mount(s"s3a://$AccessKey:$SecretKey@$AwsBucketName", s"/mnt/$MountName", "sse-s3")
dbutils.fs.mount(s"s3a://$AccessKey:$SecretKey@$AwsBucketName", s"/mnt/$MountName", "sse-kms")
dbutils.fs.mount(s"s3a://$AccessKey:$SecretKey@$AwsBucketName", s"/mnt/$MountName", "sse-kms:$KmsKey")
```

That last parameter makes me think that `databricks_aws_s3_mount` resource should have an attribute for setting that encryption. That would also be an easy way for me to configure this. 

A less desirable outcome is to do _nearly everything_ in Terraform and then have to run custom notebooks and init scripts and AWS / spark configs manually. ~If it did have to be init scripts then if those could be source-controlled and automated along with the terraform deployment that might be OK.~ ( EDIT: the [databricks_global_init_script](https://registry.terraform.io/providers/databrickslabs/databricks/latest/docs/resources/global_init_script) solves that part. )

I can enforce encryption on the buckets using the AWS terraform provider resources, and also allow instance profile to use the encryption key, ~but then either the AWS configurations setting OR init script would need to be run,~ then use the init script resource which only leaves the buckets mounted with encryption which looks like it has to be run in a notebook for now.

**Comment by alexott** [comment]
`databricks_aws_s3_mount` will be obsolete after #497 is implemented, so it should be done there. I need to investigate if it would be possible to specify encryption pieces via extra_options or it requires the third argument to the `dbutils.fs.mount`

**Comment by nfx** [comment]
@alexott it's a third argument over there

**Comment by nfx** [comment]
closing in favour of #497

**Comment by alexott** [comment]
It’s in my todo list

**Comment by alexott** [comment]
should be fixed by #497, please test

**Comment by alexott** [comment]
should be fixed by #737
